lib/phoenix/endpoint/cowboy2_adapter.ex

```diff
@@ -80,6 +80,9 @@ defmodule Phoenix.Endpoint.Cowboy2Adapter do
       Application.ensure_all_started(:ssl)
     end
 
+    # Cures a Cowboy race condition where it doesn't see our declared websocket_init/1
+    _ = Code.ensure_loaded?(Phoenix.Endpoint.Cowboy2Handler)
```

Comment:
This was reliably buggy on the server's first WebSocket upgrade without this - switching the handler within Cowboy seems to cause a race with the new handler not being available right away, and so the new handler's (optional!) websocket_init/1 was not being called.

Since this isn't a module that never gets reloaded we're OK having this just load once at startup.

Reply:
I can see why this happens: we should fix it in Plug.Cowboy. We should call `handler.module_info(:module)` before passing it to the websocket upgrade. :)

Reply:
I have republished v2.6.0. We should be able to remove this line.

Reply:
Confirmed that this fixes the issue. Branch updated

Reply:
Just so I understand for next time, does that change fix it due to the side effect of ensuring the module is in fact loaded before calling `upgrade/5`?